apollo-router/src/services/router/service.rs

```diff
@@ -916,13 +912,33 @@ impl RouterCreator {
         // For now just call activate to make the gauges work on the happy path.
         apq_layer.activate();
 
-        Ok(Self {
-            supergraph_creator,
-            static_page,
+        let router_service = content_negotiation::RouterLayer::default().layer(RouterService::new(
+            supergraph_creator.clone(),
             apq_layer,
-            query_analysis_layer,
             persisted_query_layer,
-            batching: configuration.batching.clone(),
+            query_analysis_layer,
+            configuration.batching.clone(),
+        ));
+
+        // NOTE: This is the start of the router pipeline (router_service)
+        let sb = ServiceBuilder::new()
+            // NOTE: Buffer is required because the static_page service is not Clone.
+            // The buffer should be >= concurrency as per the advice
+            // at: https://docs.rs/tower/latest/tower/buffer/struct.Buffer.html#a-note-on-choosing-a-bound
+            .buffer(50_000)
```

Comment:
Can't we just apply the buffer only on static_page ?

Reply:
Probably. I'm also hopeful that we could modify static_page to be `Clone` and avoid the Buffer completely. We aren't quite at that stage of the work yet. Still trying to get tests to pass.

Reply:
I looked into this at some length and it's actually SyncCheckpoint which we would need to modify to be `Clone` and we can't, because if we did, we would need to modify all our plugin interfaces to be `BoxCloneService` rather than `BoxService`.

I think this is something we won't be able to fix in router 2.0.